### PR TITLE
Add more goverance wording to the contribution.md

### PR DESCRIPTION
**Pull request by AloisReitbauer**
Please review, I took standard open source goverance rules and adde them to our contributing md. @plehegar please provide input from a W3C perspective.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,55 @@
-# Contributing into Distributed Trace Context
+# Governance and Contribution to the Distributed Tracing working group
+
+This document describes the contribution and governance rules for the [W3C Distributed Tracing Working Group](https://www.w3.org/2018/distributed-tracing/). This includes the following specifications
+
+- [Trace Context](https://github.com/w3c/trace-context)
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md:5]: why not correlation context?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,6 +1,55 @@
...
+
+## Meetings
+In order to provide equitable rights to all members, the following process will be followed:
+
+- Official conference calls are held on a bi-weekly basis and are announce on a [shared calendar](https://calendar.google.com/calendar/embed?src=dynatrace.com_5a09qhua6fh7jb23h7vdjg6veg%40group.calendar.google.com)
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: announce => announced

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,6 +1,55 @@
...
+- Official face-to-face meetings will be announced at least 4 weeks in advance.
+- Proposed changes to any document will be done via a Pull Request (PR).
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: *Suggestion*: formulate it in present time and make accent that this is the only way to make a change. Like "Changes to the document MUST be submitted as Pull Requests after initial discussion in form of GitHub issue or during the meeting."

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,6 +1,55 @@
...
+- PRs will be reviewed during official meetings, but off-line reviews (LGTMs, NOT LGTMs) and comments are strongly encouraged to gauge the group's opinion on the proposed change prior to the meeting.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: mention issues for initial discussion as well.

Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: There should be process for PRs to be merged without the meeting. Especially for smaller ones

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,6 +1,55 @@
...
+- During meetings, priority will be given to PRs that appear to be ready for a vote over those that appear to require discussions.
+- PRs should not be merged if they have had substantial changes made within two days of the meeting. Rebases, typo fixes, etc. do not count as substantial. Note, administrivia PRs that do not materially modify output documents may be processed by admins as needed.
+- Resolving PRs ("merging" or "closing with no action") will be done as a result of a motion made during a meeting, and approved.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: Should there be definition of quorum?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,6 +1,55 @@
...
+- Reopening a PR can be done if new information is made available, and a motion to do so is approved.
+- Any motion that does not have "unanimous consent" will result in a formal vote. See Voting.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: again, there should be a definition of quorum

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,6 +1,55 @@
...
+
+
+## Participation
+There are three categories of project participation:
+
+1 - Working Group Member. This is anyone who officially participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). Working group members can file PRs against documents and can participate in the PR voting procedure. Working Group members need to be employed by a W3C member organisation or participate as invited experts.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: what indices signifies? Is it's a list - should be a dot after index.

Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: organisation  -> organization

Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: should this "W3C member organisation" be "W3C member organization that officially joined the working group"

Comment (SergeyKanzhelev) [reply CONTRIBUTING.md]: Also, suggestion I'd lead with this as the beginning of this item suggest general participation when the idea of the item is that you have to join officially

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,6 +1,55 @@
...
+- Resolving PRs ("merging" or "closing with no action") will be done as a result of a motion made during a meeting, and approved.
+- Reopening a PR can be done if new information is made available, and a motion to do so is approved.
+- Any motion that does not have "unanimous consent" will result in a formal vote. See Voting.
+
+
+## Participation
+There are three categories of project participation:
+
+1 - Working Group Member. This is anyone who officially participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). Working group members can file PRs against documents and can participate in the PR voting procedure. Working Group members need to be employed by a W3C member organisation or participate as invited experts. 
+2 - Non Working Group Member. This is anyone who  participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). In contrast to Working Group members they cannot file a PR or vote on a PR. They can, however, create issues for specifcations.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: > cannot file a PR

How can we enforce it and should we?

Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: Please install spell checker. "specifcations" -> "specifications"

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,6 +1,55 @@
...
+
+
+## Participation
+There are three categories of project participation:
+
+1 - Working Group Member. This is anyone who officially participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). Working group members can file PRs against documents and can participate in the PR voting procedure. Working Group members need to be employed by a W3C member organisation or participate as invited experts. 
+2 - Non Working Group Member. This is anyone who  participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). In contrast to Working Group members they cannot file a PR or vote on a PR. They can, however, create issues for specifcations. 
+3 - Chairs and Co-Chairs are members of the group but have the ability to perform administrative actions on behalf of the group. For example, manage the website, github repos and moderate the meetings. Their actions should be done with the knowledge and consent of the group. They also have the ability to merge/close PRs, but only per the group's approval.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: I'd suggest to expand on "consent" part of this

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,6 +1,55 @@
...
+2 - Non Working Group Member. This is anyone who  participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). In contrast to Working Group members they cannot file a PR or vote on a PR. They can, however, create issues for specifcations. 
+3 - Chairs and Co-Chairs are members of the group but have the ability to perform administrative actions on behalf of the group. For example, manage the website, github repos and moderate the meetings. Their actions should be done with the knowledge and consent of the group. They also have the ability to merge/close PRs, but only per the group's approval. 
+
+## PRs
+Typically, PRs are expected to meet the following criteria prior to being merged:
+
+- The author of the PR indicates that it is ready for review by asking for it to be discussed in an upcoming meeting - eg. by adding it to the agenda document.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: where is agenda document?

Comment (SergeyKanzhelev) [reply CONTRIBUTING.md]: Can there be some other gesture on PR to suggest readiness. It's unrealistic to expect people to keep track of agenda document before reviewing anything

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,6 +1,55 @@
...
+- All comments have been addressed.
+- PRs that have objections/concerns will be discussed off-line by interested parties. A resolution, updated PR, will be expected from those talks.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: > off-line

please expand on the definition of off-line here

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,6 +1,55 @@
...
+
+## Voting
+If a vote is taken during a meeting, the follow rules will be followed:
+
+- There is only 1 vote per participating company, or nonaffiliated individual.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: is it a typical practice? Should the vote of implementer be bigger than party that hasn't implemented? Are only invited nonaffiliated individuals participate? Right now it's only Yuri

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -1,6 +1,55 @@
...
+- Each participating company can assign a primary and secondary representative.
+A participating company attains voting rights by having any of the entity's assigned representative(s), in aggregate, attend 3 out of the last 4 meetings. The people listed as "primary" or "alternate" for an entity can be changed no more than once a month and can be changed by notifying one of the admins. The entity obtains voting rights after the 3rd meeting, not during.
+- Only members with voting rights will be allowed to vote.
+- A vote passes if more than 50% of the votes cast approve the motion.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: is 50% a typical practice? 70% sounds better for me in terms of guaranteeing better adoption and faireness

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -1,6 +1,55 @@
...
+- The author of the PR indicates that it is ready for review by asking for it to be discussed in an upcoming meeting - eg. by adding it to the agenda document.
+- All comments have been addressed.
+- PRs that have objections/concerns will be discussed off-line by interested parties. A resolution, updated PR, will be expected from those talks.
+
+## Voting
+If a vote is taken during a meeting, the follow rules will be followed:
+
+- There is only 1 vote per participating company, or nonaffiliated individual.
+- Each participating company can assign a primary and secondary representative.
+A participating company attains voting rights by having any of the entity's assigned representative(s), in aggregate, attend 3 out of the last 4 meetings. The people listed as "primary" or "alternate" for an entity can be changed no more than once a month and can be changed by notifying one of the admins. The entity obtains voting rights after the 3rd meeting, not during.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: I'd suggest to keep a member if participation is online-only. I.e. if company rep actively commenting on PRs and issues and submitting PRs she/he should not be forced to dial in or even fly to the meeting.

**Review by SergeyKanzhelev — CHANGES_REQUESTED** [review]
I have added a few comments. Process in general feels a little bit heavy. Maybe reasonably heavy, but I'd suggest there should be simplified version of it with simple flow: Issue -> PR -> X approvals with listed name/company affiliation -> Merge. And all the heavy process with casting vote and discussion during the meeting is for the PRs that cannot collect enough approvals. This way we open door for minor contributions and close faster on obvious issues. And allow more time for difficult decision discussion during the meeting.

**Comment by yurishkuro** [comment]
> Issue -> PR -> X approvals with listed name/company affiliation -> Merge. And all the heavy process with casting vote and discussion during the meeting is for the PRs that cannot collect enough approvals.

I agree with that. Currently the problem is that so far some PRs have been merged without getting hardly any approvals (i.e. 1). A simple majority vote "+1 binding" / "+1 non-binding" would address my concerns in https://github.com/w3c/trace-context/issues/165 (btw, please move that issue here). If there is a situation when people fail to vote and even simple PRs are getting stuck, then a time frame can be established for "speak now or hold your peace".

File: CONTRIBUTING.md (hunk 5 above)
Comment (mtwo) [reply CONTRIBUTING.md]: I think that quorum is defined further down

File: CONTRIBUTING.md (hunk 12 above)
Comment (mtwo) [reply CONTRIBUTING.md]: Adding on to this - does the W3C typically give each committee member a vote or each sponsoring company? We should follow whatever the rest of the W3C does

File: CONTRIBUTING.md (hunk 13 above)
Comment (mtwo) [reply CONTRIBUTING.md]: We should follow whatever the rest of the W3C does

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -1,6 +1,55 @@
...
+- Voting process:
+   - Comment on the PR: "YES VOTE", "NO VOTE", or "ABSTAIN".
+   - Any person is encouraged to vote with a statement of support or dissent to help undecided voters to reach a decision
+   = Comments are welcome from non-members
+   = Voting tally will reflect the above qualification and recorded in PR
+
 
-This repository is being used for work in the [W3C Distributed Trace Context Working Group](https://www.w3.org/2018/distributed-tracing/).
 
 ## Editing report
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: btw, this section can be replaced by "see contributing guide of specification repository" as there is no reports in this repository.

File: CONTRIBUTING.md (hunk 1 above)
Comment (AloisReitbauer) [reply CONTRIBUTING.md:5]: I can add it. Just did not as we are not technically working on it. I just added it to the spec

File: CONTRIBUTING.md (hunk 3 above)
Comment (AloisReitbauer) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 14 above)
Comment (yurishkuro) [reply CONTRIBUTING.md]: +1 to what @SergeyKanzhelev said, given that it's impossible to find a time for a meeting convenient for every possible member across the globe.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -1,6 +1,56 @@
...
+
+## PRs
+Typically, PRs are expected to meet the following criteria prior to being merged:
+
+- The author of the PR indicates that it is ready for review by asking for it to be discussed in an upcoming meeting - eg. by adding it to the agenda document.
+- All comments have been addressed.
+- PRs that have objections/concerns will be discussed off-line by interested parties. A resolution, updated PR, will be expected from those talks.
+
+## Voting
+If a vote is taken during a meeting, the follow rules will be followed:
```
Comment (yurishkuro) [inline CONTRIBUTING.md]: -1 to any votes during meeting, instead of explicit votes on PRs or mailing list.

File: CONTRIBUTING.md (hunk 14 above)
Comment (mtwo) [reply CONTRIBUTING.md]: +1

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -1,6 +1,56 @@
...
+- PRs should not be merged if they have had substantial changes made within two days of the meeting. Rebases, typo fixes, etc. do not count as substantial. Note, administrivia PRs that do not materially modify output documents may be processed by admins as needed.
+- Resolving PRs ("merging" or "closing with no action") will be done as a result of a motion made during a meeting, and approved.
+- Reopening a PR can be done if new information is made available, and a motion to do so is approved.
+- Any motion that does not have "unanimous consent" will result in a formal vote. See Voting.
+
+
+## Participation
+There are three categories of project participation:
+
+1. Non Working Group Member. This is anyone who  participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). In contrast to Working Group members they can file a PR but not vote on a PR. They can, however, create issues for specifcations.
```
Comment (plehegar) [inline CONTRIBUTING.md]: participation is subject to https://www.w3.org/2018/Process-20180201/#Policies . As such, while we accept and encourage comments and contributions in mailing lists and GH, meeting participants are subject to https://www.w3.org/2018/Process-20180201/#GeneralMeetings .

File: CONTRIBUTING.md (hunk 16 above)
Comment (plehegar) [reply CONTRIBUTING.md]: -1 as well: our charter says: "To afford asynchronous decisions and organizational deliberation, any resolution (including publication decisions) taken in a face-to-face meeting or teleconference will be considered provisional. A call for consensus (CfC) will be issued for all resolutions (for example, via email and/or web-based survey), with a response period from one week to 10 working days, depending on the chair's evaluation of the group consensus on the issue. If no objections are raised on the mailing list by the end of the response period, the resolution will be considered to have consensus as a resolution of the Working Group. ".

File: CONTRIBUTING.md (hunk 12 above)
Comment (plehegar) [reply CONTRIBUTING.md]: Voting requirements are listed in https://www.w3.org/2018/Process-20180201/#Votes .

File: CONTRIBUTING.md (hunk 14 above)
Comment (plehegar) [reply CONTRIBUTING.md]: Our charter doesn't restrict the number of individuals from a participating Member so I can't think we can put a limit outside the charter, per "A group charter may limit the number of individuals representing a W3C Member (or group of related Members)." https://www.w3.org/2018/Process-20180201/#member-rep

File: CONTRIBUTING.md (hunk 13 above)
Comment (plehegar) [reply CONTRIBUTING.md]: W3C Groups rarely conducts formal votes nowadays, so there is no typical number. At the minimum, I suggest adding a quorum requirement before conducting a vote.

**Comment by plehegar** [comment]
Modulo making sure we don't conflict with the W3C Process and the Working Group charter (see comments), this seems fine to me. I'm worried about the need to do votes however, which is unusual nowadays for a W3C Group. so I'm wondering about the motivation to expand on those here. Otherwise, keep in mind that the W3C Process always prevail, then the charter, then this CONTRIBUTING document. We'll also need to add a link back here from all of the CONTRIBUTING files we have in the spec repos.

**Comment by yurishkuro** [comment]
@plehegar on the voting, I see your point, it's probably too heavy process. Instead, there can be a mandatory waiting period, if no objections raised then no vote is required.

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -1,11 +1,38 @@
...
+- Official face-to-face meetings will be announced at least 4 weeks in advance.
+- Proposed changes to any document MUST be handled via a Pull Request (PR). If the change came up during a meeting the meeting owner is responsible for creating the PR.
+- PRs will be reviewed during official meetings, but off-line reviews (LGTMs, NOT LGTMs) and comments are strongly encouraged to gauge the group's opinion on the proposed change prior to the meeting.
+- If a topic is not defined specifically enough yet to justify a PR against the specification, contributors are encouraged to file an issue. 
+- During meetings, priority will be given to PRs that appear to be ready for a vote over those that appear to require discussions.
+- PRs should not be merged if they have had substantial changes made within two days of the meeting. Rebases, typo fixes, etc. do not count as substantial. Note, administrivia PRs that do not materially modify output documents may be processed by admins as needed.
```
Comment (iredelmeier) [inline CONTRIBUTING.md]: typo: administrivia -> administrative

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -1,11 +1,49 @@
...
+2. Working Group Member. This is anyone who officially participates in the group's activities in any of our communication channels (email, github issues/PRs, meetings, etc.). Working group members can file PRs against documents and can participate in the PR voting procedure. Working Group members need to be employed by a W3C member organization or participate as invited experts. The [W3C process document](https://www.w3.org/2018/Process-20180201/#MemberBenefits) defines the rights of members. 
+3. Chairs and Co-Chairs are members of the group but have the ability to perform administrative actions on behalf of the group. For example, manage the website, github repos and moderate the meetings. Their actions should be done with the knowledge and consent of the group. They also have the ability to merge/close PRs, but only per the group's approval. 
+
+## PRs and Voting
+
+Specificaiton changes are proposed via PRs which are expected to meet the following criteria prior to being merged:
+
+- The author of the PR indicates that it is ready for review by asking for it to be discussed in an upcoming meeting - eg. by adding it to the agenda document.
```
Comment (SergeyKanzhelev) [inline CONTRIBUTING.md]: Perhaps there might be another form of indication like remove `[WIP]` from the title?

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
@AloisReitbauer one minor feedback on how to indicate that PR is ready for review. Perhaps this should be assumed unless there is a `[WIP]` in the title?

**Review by mtwo — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
should this be merged?

**Comment by AloisReitbauer** [comment]
Merging as there are no open comments and it has been open for a while.

**Pull request merged**